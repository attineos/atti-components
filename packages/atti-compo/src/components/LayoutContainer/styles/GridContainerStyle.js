import styled from 'styled-components'

import { join, range, map } from 'lodash'

import wrapMediaQuery from 'helpers/wrapMediaQuery'
import BaseContainerStyle from './BaseContainerStyle'

const getColumnTemplate = colCount => map(range(colCount), () => '1fr').join(' ')

const getColumnTemplateForIE = (colCount, gapSize) =>
  map(range(colCount), () => '1fr').join(` ${gapSize}px `)

const getStyleString = (colCount, totalSize, gapSize) => `
  grid-gap: ${gapSize};
  max-width: ${totalSize}
  -ms-grid-columns: ${getColumnTemplateForIE(colCount, gapSize)};
  grid-template-columns: ${getColumnTemplate(colCount)};
`

const GridContainerStyle = styled(BaseContainerStyle)`
  display: -ms-grid;
  display: grid;

  ${({ theme }) =>
    join(
      map(theme.layout, layout =>
        wrapMediaQuery(
          layout.from,
          layout.to,
          getStyleString(layout.cols, layout.maxWidth, layout.gap),
        ),
      ),
      '',
    )};
`

export default GridContainerStyle
