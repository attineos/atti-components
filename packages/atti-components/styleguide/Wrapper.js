import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
