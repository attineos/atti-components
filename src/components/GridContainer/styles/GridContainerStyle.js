import styled from 'styled-components'

import { range, map } from 'lodash'

const getColumnTemplate = colCount => map(range(colCount), () => '1fr').join(' ')

const getColumnTemplateForIE = (colCount, gapSize) =>
  map(range(colCount), () => '1fr').join(` ${gapSize}px `)

const getStyleString = (colCount, totalSize, gapSize) => `
grid-gap: ${gapSize};
max-width: ${totalSize}
-ms-grid-columns: ${getColumnTemplateForIE(colCount, gapSize)};
grid-template-columns: ${getColumnTemplate(colCount)};
`

const GridContainerStyle = styled.div`
  display: -ms-grid;
  display: grid;

  width: 100%;

  ${({ theme }) =>
    getStyleString(
      theme.components.grid.cols.xlDesktop,
      theme.components.grid.maxWidth.xlDesktop,
      theme.components.grid.gap.xlDesktop,
    )}
    
  @media (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
  ${({ theme }) =>
    getStyleString(
      theme.components.grid.cols.desktop,
      theme.components.grid.maxWidth.desktop,
      theme.components.grid.gap.desktop,
    )}
  }
  
  @media (max-width: ${({ theme }) => theme.breakpoints.tablet}) {
    ${({ theme }) =>
      getStyleString(
        theme.components.grid.cols.tablet,
        theme.components.grid.maxWidth.tablet,
        theme.components.grid.gap.tablet,
      )}
  }
  
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    ${({ theme }) =>
      getStyleString(
        theme.components.grid.cols.mobile,
        theme.components.grid.maxWidth.mobile,
        theme.components.grid.gap.mobile,
      )}
  }
`

export default GridContainerStyle
