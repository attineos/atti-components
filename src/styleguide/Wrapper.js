import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
