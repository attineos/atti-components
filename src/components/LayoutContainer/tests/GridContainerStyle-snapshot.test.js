/* This file was generated automatically, please do not write in it. */
import React from 'react';
import GridContainerStyle from '../styles/GridContainerStyle';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('GridContainerStyle renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <GridContainerStyle ></GridContainerStyle>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    