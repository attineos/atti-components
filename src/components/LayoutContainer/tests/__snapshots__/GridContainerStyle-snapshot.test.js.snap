// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridContainerStyle renders correctly 1`] = `
<span
  className="atti-box"
>
  <div
    className="GridContainerStyle-sc-14lyijq-0 fDWKy"
  />
</span>
`;
